hikari/utilities/iterators.py

```diff
@@ -630,6 +662,10 @@ def __aiter__(self) -> LazyIterator[ValueT]:
         # We are our own iterator.
         return self
 
+    def __iter__(self) -> LazyIterator[ValueT]:
+        # This iterator is async only.
+        raise TypeError(f"{type(self).__name__} is an async-only iterator, did you mean 'async for'?")
```

Comment:
```suggestion
    def __iter__(self) -> typing.NoReturn:
        # This iterator is async only.
        cls = type(self)
        raise TypeError(f"{cls.__module__}.{cls.__qualname__} is an async-only iterator, did you mean 'async for'?")
```

Reply:
If i do that we won't be able to override it with ```python
    def __iter__(self) -> FlatLazyIterator[ValueT]:
        return self
``` in the FlatLazyIterator